### PR TITLE
Prime1: Logic Update

### PR DESCRIPTION
**Pull request by Samuel6710**
##### Chozo Ruins

- Added: Reverse Flaahgra in Sun Tower is now logical
- Added: Furnace E Tank Wall Boost Escape
- Added: Transport Access North Wallboost to Hive Totem from Elevator
- Added: Trigger Ghosts from Sun Tower Access without Bombs or Spider

##### Phazon Mines

- Added: Fungal Hall A now has Energy and Combat Logic
- Added: Fungal Hall A SJ Scan Dash Grapple Skip
- Added: Fungal Hall Access NSJ Bombless Escape to Fungal Hall A

##### Phendrana Drifts

- Changed: Phendrana Canyon Pickup NSJ Bombless Triple Boost Adjustments
- Added: Hunter Cave Bunny Hop to reach Hunter Cave Access from Lower Edge Tunnel
- Added: Hunter Cave Slope Jump to reach Chamber Access from Lake Tunnel

##### Tallon Overworld

- Added: Root Cave Climb NSJ Boost Strat

### CODE REVIEW

**Comment by toasterparty** [comment]
@henriquegemignani Is there going to be a problem with sunchamber + suntower PoNR like we had years ago?

**Comment by henriquegemignani** [comment]
> @henriquegemignani Is there going to be a problem with sunchamber + suntower PoNR like we had years ago?

Best to investigate and see what happens. Generate a bunch before and after and compare results.

File: randovania/games/prime1/json_data/Phazon Mines.txt [hunk 1]
```diff
@@ -1946,15 +1948,25 @@ Extra - aabb: [90.076614, 160.53384, -176.2045, 166.0206, 330.8736, -124.46773]
...
+                  Space Jump Boots
+                  Any of the following:
+                      R-Jump (Intermediate) or Use Grapple Beam
+                      Scan Visor and Combat/Scan Dash (Intermediate)
+              All of the following:
+                  Scan Visor
+                  Any of the following:
+                      Combat/Scan Dash (Hypermode)
+                      Combat/Scan Dash (Expert) and Use Grapple Beam
+          Any of the following:
```
Comment (toasterparty) [inline randovania/games/prime1/json_data/Phazon Mines.txt:1963]: Uncharged plasma feels pretty powerful here. At the end of the day, writing logic like this just plain sucks, so don't feel like you have to for every room.

Also remember that you are picking health values that come before the normal 1.5x scaling randovania does by default.

File: randovania/games/prime1/json_data/Chozo Ruins.txt [hunk 2]
```diff
@@ -1267,6 +1267,8 @@ Extra - aabb: [203.54953, 323.97684, 5.845402, 266.79828, 366.83566, 99.515015]
               Morph Ball Bomb and Complex Bomb Jump (Intermediate) and Normal Damage ≥ 20
               # Roll down to the gap in the track and back up to trigger ghosts, skipping super missiles. This connection makes fighting Flaahgra without Spider Ball a dangerous action.
               Spider Ball
+              # https://youtu.be/pEruOGbEteI
+              Instant Unmorph Jump (Hypermode)
```
Comment (toasterparty) [inline randovania/games/prime1/json_data/Chozo Ruins.txt:1271]: my boi you need sj

**Review by toasterparty — CHANGES_REQUESTED** [review]

File: randovania/games/prime1/json_data/Phendrana Drifts.txt [hunk 3]
```diff
@@ -1997,6 +2002,9 @@ Extra - aabb: [185.53804, -1270.4935, -14.424318, 272.45685, -1190.654, 37.42458
               Any of the following:
                   Gravity Suit
                   Slope Jump (Intermediate) and Gravityless Underwater Movement (Intermediate)
+  > Door to Chamber Access
+      # https://youtu.be/oPvUjPIBQQo?t=27
```
[the remark this answers is not in the document]
Comment (Samuel6710) [reply randovania/games/prime1/json_data/Phendrana Drifts.txt:2006]: I tried it out myself. The collision of the ledge is misleading, the standable part of the slope extends a lot more than it looks like it does.

**Comment by Samuel6710** [comment]
I'll be able to get these changes in later today.

File: randovania/games/prime1/json_data/Phendrana Drifts.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (toasterparty) [reply randovania/games/prime1/json_data/Phendrana Drifts.txt:2006]: all good

File: randovania/games/prime1/json_data/Phazon Mines.txt (hunk 1 above)
Comment (toasterparty) [reply randovania/games/prime1/json_data/Phazon Mines.txt:1963]: You increased the numbers but my concern was that they were too high. How did you test these numbers? They should represent the best possible scenario, not the least.

File: randovania/games/prime1/json_data/Tallon Overworld.txt [hunk 4]
```diff
@@ -543,11 +543,7 @@ Extra - aabb: [-703.2118, 297.52924, -49.04736, -619.3062, 378.77765, 64.30938]
   > Door to Transport Tunnel B
       Trivial
   > Door to Arbor Chamber
-      All of the following:
-          Boost Ball and Morph Ball and Power Bomb and Movement (Expert) and Single Room Out of Bounds (Advanced) and Standable Terrain (Advanced)
-          Any of the following:
-              Instant Unmorph Jump (Expert) and Wall Boost (Hypermode)
-              Morph Ball Bomb and Bomb Space Jump (Advanced)
+      Trivial
```
Comment (toasterparty) [inline randovania/games/prime1/json_data/Tallon Overworld.txt:546]: Trivial? 🙃

**Review by toasterparty — CHANGES_REQUESTED** [review]

File: randovania/games/prime1/json_data/Tallon Overworld.txt (hunk 4 above)
Comment (Samuel6710) [reply randovania/games/prime1/json_data/Tallon Overworld.txt:546]: ??

Comment (Samuel6710) [reply randovania/games/prime1/json_data/Tallon Overworld.txt:546]: Oh I see what I did wrong

**Comment by Samuel6710** [comment]
@toasterparty Let me know what you think :)

**Review by toasterparty — APPROVED** [review]

**Pull request merged**